app/models/authentication.rb

```diff
@@ -1,11 +1,81 @@
 #encoding: utf-8
 class Authentication < ActiveRecord::Base
+  before_destroy :can_be_destroyed?, prepend: true

```

Comment:
Qui va' sicuramente eliminato.
Nel caso in cui ci siano dei dependent: destroy valuta la condizione prima che vengano distrutti (e quindi evita che vengano distrutti), però sembra che sia obsoleto e che questo comportamento in Rails 4.1 sia già il default.
https://github.com/bbatsov/rails-style-guide/issues/120